pingcap/tidb-operator
# stop backup job before cleanup `Backup`

### What problem does this PR solve?

Fix #4099
### What is changed and how does it work?

wait for backup job to finish before clean job is created. If the backup job is running, will delete it.

### Code changes


### Tests


* Delete a `Backup` when the backup job is running
   1. Deploy a `Backup`.
   2. Delete the `Backup` when the backup job is running.
   3. Check if the backup job is deleted before the cleanup job is created.
   4. Check if the cleanup job is created.
   5. Check all data in storage is deleted.
* Delete a  `Backup` when the backup job is completed
   1. Deploy a `Backup` and wait for it to be completed.
   2. Delete the `Backup`.
   3. Check if the backup job isn't deleted and the cleanup job is created.
   4. Check all data in storage is deleted.

### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Fix the issue that backup data maybe remain after cleaning running backup
```
